app/models/lockstep/magic_link_summary.rb

```diff
@@ -0,0 +1,16 @@
+class Lockstep::MagicLinkSummary < Lockstep::ApiRecord
+  self.model_name_uri = "v1/useraccounts/magic-links/summary"
+  self.id_ref = "group_key"
+  self.query_path = ""
+  load_schema(Schema::MagicLinkSummary)
+
+  def self.with_date(from_date: nil, to_date: nil)
+  	resp = resource.get('', params: { from: from_date, to: to_date })
+    raise Lockstep::Exceptions::BadRequestError, 'Endpoint not found' if resp.code == '404'
+    raise StandardError.new("#{resp.code} error while fetching: #{resp.body}") unless %w(201 200 204).include?(resp.code.to_s)
+
+    summary = JSON.parse(resp.body)
```

Comment:
In case of a 204, resp.body would be nil. Should we handle that check? If we never expect a 204 from platform, let's remove that as a part of condition from Line 10.